source/include/FreeRTOS_DNS_Callback.h

```diff
@@ -27,37 +27,46 @@
 
 
 #ifndef FREERTOS_DNS_CALLBACK_H
-#define FREERTOS_DNS_CALLBACK_H
+    #define FREERTOS_DNS_CALLBACK_H
 
-#if ( ( ipconfigDNS_USE_CALLBACKS == 1 ) && ( ipconfigUSE_DNS != 0 ) )
+    #ifdef __cplusplus
+        extern "C" {
+    #endif
 
 /* FreeRTOS includes. */
     #include "FreeRTOS.h"
 
 /* FreeRTOS+TCP includes. */
     #include "FreeRTOS_IP.h"
+    #include "FreeRTOSIPConfigDefaults.h"
```

Comment:
`FreeRTOS_IP.h` already includes the `FreeRTOSIPConfigDefaults.h` file, why do we need to include it separately?